lib/cog/command/pipeline/executor.ex

```diff
@@ -149,36 +117,51 @@ defmodule Cog.Command.Pipeline.Executor do
     end
   end
 
-  @doc """
-  `parse` -> {:stop, :shutdown} | {:next_state, :bind}
-
-  After connecting the request string is parsed and an invocation
-  (`%Piper.Ast.Invocation{}`) or a pipeline(`%Piper.Ast.Pipeline{}`) is added
-  to `remaining` in the state.
-  """
   def parse(:timeout, state) do
-    case Parser.scan_and_parse(state.request["text"], command_resolver: &CommandResolver.find_bundle(&1, state.user), return_pipeline: true) do
+    options = %ParserOptions{resolver: CommandResolver.command_resolver_fn(state.user)}
+    case Parser.scan_and_parse(state.request["text"], options) do
```

Comment:
Do we have any plans to add a struct for request? Seems weird that we're indexing into strings in a lot of places.